rules-tests/Strict/Rector/ClassMethod/AddConstructorParentCallRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rector\Tests\Strict\Rector\ClassMethod\AddConstructorParentCallRector\Fixture;
+
+use Rector\Tests\Strict\Rector\ClassMethod\AddConstructorParentCallRector\Source\ParentClassWithConstructor;
+
+class SunshineCommand extends ParentClassWithConstructor
+{
+    public function __construct()
```

Comment:
Could you add new fixture to skip if it has parameter already? Because the usage of parameter in child can be different

```php
public function __construct(DifferentParamType $param)
```

see https://3v4l.org/MTsnC

Reply:
I don't think it should skip as PHPStan still errors as there's no call to parent still (https://phpstan.org/r/34e37898-4200-4259-856f-bca40810f3f8).

However, I've added a new fixture for the case of params of same name but different type.

Looks like there was a bug where it would add the new param as expected but it would be invalid PHP code because the names would be the same. You'd end up with two params with the same name and different types. I've fixed it so it adds an increment on the new param name. This bug would affect `GetToConstructorInjectionRector` also.

Reply:
I prefer to just skip that as I think the usage may vary, `__construct()` usage is very fragile on this case IMO, but I will let @TomasVotruba decide on this

Reply:
@samsonasik I think we can merge it as it is and iterate from there. Different parameter type is a different check of PHPStsan.

Reply:
@TomasVotruba Ok then 👍